Xenakios/TakeRenaming.cpp

```diff
@@ -223,40 +165,39 @@ void DoRenameTakeAndSourceFileDialog(COMMAND_T* ct)
 					if (newfnsplit[2].compare(fnsplit[2])!=0)
 						newfilename.append(fnsplit[2]);
 
-					MoveFile(oldname.c_str(),newfilename.c_str());
+					if (FileOrDirExists (newfilename.c_str()))
+					{
+						if (MessageBox(g_hwndParent, __LOCALIZE("Filename already exists or file could not be offlined.\nIf you continue it might create a wrong source reference in your REAPER project!\nContinue?", "sws_DLG_165"), __LOCALIZE("Xenakios - Warning", "sws_DLG_165"), MB_OKCANCEL) == IDCANCEL)
+						{
+							Main_OnCommand(40101, 0); // online all media
+							continue;
+						}					
+					}
 					AddToRenameLog(oldname,newfilename);
 
-					for (int j=0;j<(int)alltakes.size();j++)
+					for (size_t j=0;j<alltakes.size();j++)
 					{
 						PCM_source *thesrc=(PCM_source*)GetSetMediaItemTakeInfo(alltakes[j],"P_SOURCE",0);
-						if (thesrc && thesrc->GetFileName() && strcmp(thesrc->GetType(),"SECTION")!=0)
+						if (thesrc)
 						{
-							string fname;
-							fname.assign(thesrc->GetFileName());
-							if (oldname.compare(fname)==0)
+							if (!strcmp(SWS_GetSourceFileName(thesrc), oldname.c_str()))
```

Comment:
These two conditions can be merged with `&&` to save an indentation level. Or two levels by reversing them:

```cpp
if (!thesrc || strcmp(...))
  continue;
```